src/metabase/db/data_source.clj

```diff
@@ -15,6 +15,8 @@
 
 (set! *warn-on-reflection* true)
 
+;; NOTE: do not create DataSource directly, used one of our helper functions instead
+;; we want to make sure [[update-h2/update-if-needed!]] is called on every data source
```

Comment:
I think this grammar could be clearer, how about this?

> Never instantiate a DataSource directly - always use a helper function to ensure [[update-h2/update-if-needed!]] is called.

Reply:
Also it would be nice to directly reference the factory methods they can use, so they don't need to look around.